streuseval.py

```diff
@@ -110,8 +112,8 @@ class Ratio(object):
     (fractions.Fraction reduces e.g. 378/399 to 18/19. We want to avoid this.)
     '''
     def __init__(self, numerator, denominator):
-        self._n = numerator
-        self._d = denominator
+        self._n = float(numerator) if isinstance(numerator, Ratio) else numerator
+        self._d = float(denominator) if isinstance(numerator, Ratio) else denominator
```

Comment:
What was triggering the error that makes this necessary?

Reply:
[This line](https://github.com/nert-nlp/streusle/blob/master/streuseval.py#L507), with `a='aid', k='MWE', g='Tags', h='LinkAvg', m='P'`.
It was set by [this line](https://github.com/nert-nlp/streusle/blob/master/streuseval.py#L382) to `Ratio(avg*denom, denom)`, where `avg` was already a `Ratio`.

Reply:
So the comment at https://github.com/nert-nlp/streusle/blob/56c131473e6a85e4095151971dc8453f5299822e/streuseval.py#L379 is incorrect? Would it be better to have it as `avg = float((scores[k]['Link+'][m]+scores[k]['Link-'][m])/2)`?

Reply:
I think your line 116 has a bug—should be `isinstance(denominator, Ratio)`. But ideally we'd actually be able to avoid converting to float. Since `(a/b) / (c/d) == (a*d)/(b*c)`, how about:

```
    def __init__(self, numerator, denominator):
        self._n = numerator.numerator if isinstance(numerator, Ratio) else numerator
        if isinstance(denominator, Ratio):
            self._n *= denominator.denominator
        self._d = denominator.numerator if isinstance(denominator, Ratio) else denominator
        if isinstance(numerator, Ratio):
            self._d *= numerator.denominator
```

Reply:
Yes, looks like that comment is incorrect.
I don't know if it's better to convert `avg` to float or to change the constructor to what you suggested. They both look fine. I just don't want any exceptions when I'm scoring :)
I just changed it to a version equivalent to your suggestions but with one less `isinstance` check.

Reply:
Let's try it without converting `avg` to float (unless it's necessary for it to print nicely). I think that preserves more information.

Reply:
Done.